readthedocs/builds/forms.py

```diff
@@ -33,6 +41,34 @@ class Meta(object):
         model = Version
         fields = ['active', 'privacy_level', 'tags']
 
+    def clean_active(self):
+        active = self.cleaned_data['active']
+        if self._is_default_version and not active:
+            msg = (
+                '{} is the default version of the project, '
+                'it should be active.'
+            )
+            raise forms.ValidationError(
+                _(msg.format(self.instance.verbose_name))
+            )
+        return active
+
+    def clean_privacy_level(self):
+        privacy_level = self.cleaned_data['privacy_level']
+        if self._is_default_version and privacy_level != PUBLIC:
+            msg = (
+                '{} is the default version of the project, '
+                'it should be public.'
+            )
+            raise forms.ValidationError(
+                _(msg.format(self.instance.verbose_name))
+            )
+        return privacy_level
```

Comment:
This logic doesn't hold up on our commercial hosting. If private repositories are allowed on the instance, then we should allow any value here.

Reply:
I just fixed the test that was failing because of the privacy restrictions :cry: